server/container_create.go

```diff
@@ -850,6 +850,29 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
 		specgen.AddBindMount("/etc/hosts", "/etc/hosts", options)
 	}
 
+	// bind mount default user defined secrets
+	defaultMounts, err := secretMounts(mountLabel, defaultMountCfgFile)
+	if err != nil {
+		return nil, err
+	}
+	containerMounts := containerConfig.GetMounts()
+	for _, m := range defaultMounts {
+		// if m has already been mounted, do not mount secrets
+		if isAlreadyMounted(containerMounts, m.Destination) {
+			logrus.Warn("%q has already been mounted; cannot override mount", m.Destination)
+		} else {
+			specgen.AddBindMount(m.Source, m.Destination, nil)
+		}
+	}
+	// bind mount overriden user defined secrets
+	overrideMounts, err := secretMounts(mountLabel, overrideMountCfgFile)
+	if err != nil {
+		return nil, err
+	}
+	for _, m := range overrideMounts {
+		specgen.AddBindMount(m.Source, m.Destination, nil)
```

Comment:
the logic should check if this is mounted already as it's done above

Reply:
I thought we wanted to mount it even if it has already been mounted as it is a override?

Reply:
NO, We don't want to mount it if it is mounted by the caller.  We should only mount it if, it is NOT mounted by the caller.  
We should however point out that we are not mounting it in the log with a pr_warn.

Reply:
but if it wasn't mounted on the first cycle, we shouldn't mount it here as well

Reply:
fixed.